demos/http3/ossl-nghttp3-demo-server.c

```diff
@@ -0,0 +1,635 @@
+  /* Process all the streams */
+  for (int i = 0; i < MAXSSL_IDS; i++) {
+    if (ssl_ids[i].s) {
+      item->desc = SSL_as_poll_descriptor(ssl_ids[i].s);
+      item->events = UINT64_MAX;
+      item->revents = UINT64_MAX;
+      numitem++;
+      item++;
+    }
+  }
+  int ret = SSL_poll(items, numitem, sizeof(SSL_POLL_ITEM), &nz_timeout, 0,
+                     &result_count);
+  if (!ret) {
+    fprintf(stderr, "SSL_poll failed\n");
+    return -1; /* something is wrong */
+  }
+  if (result_count == 0) {
+    /* Timeout may be something somewhere */
+    return 0;
+  }
+
+  /* We have something */
+  item = items;
+  /* SSL_accept_stream if anyway */
+  if ((item->revents & SSL_POLL_EVENT_ISB) ||
+      (item->revents & SSL_POLL_EVENT_ISU)) {
+    SSL *stream = SSL_accept_stream(ssl_ids[0].s, 0);
+    if (stream != NULL) {
+      printf("=> Received connection on %ld\n", SSL_get_stream_id(stream));
+      add_id(SSL_get_stream_id(stream), stream, h3ssl);
+      return 1; /* loop until we have all the streams */
+    }
+    return -1; /* something is wrong */
+  }
+  /* Create new streams when allowed */
+  if (item->revents & SSL_POLL_EVENT_OSB) {
+    /* at least one bidi */
+    printf("Create bidi?\n");
+  }
+  if (item->revents & SSL_POLL_EVENT_OSU) {
+    /* at least one uni */
+    printf("Create uni\n");
+    /* we have 4 streams from the client 2, 6 , 10 and 0 */
+    /* need 2 streams to the client */
+    if (!h3ssl->has_uni) {
+      printf("Create uni beacause !h3ssl->has_uni\n");
+      SSL *rstream = SSL_new_stream(ssl_ids[0].s, SSL_STREAM_FLAG_UNI);
+      if (rstream != NULL) {
+        fprintf(stderr, "=> Opened on %ld\n", SSL_get_stream_id(rstream));
+        fflush(stderr);
+      } else {
+        fprintf(stderr, "=> Stream == NULL!\n");
+        fflush(stderr);
+        return -1;
+      }
+      SSL *pstream = SSL_new_stream(ssl_ids[0].s, SSL_STREAM_FLAG_UNI);
+      if (pstream != NULL) {
+        fprintf(stderr, "=> Opened on %ld\n", SSL_get_stream_id(pstream));
+        fflush(stderr);
+      } else {
+        fprintf(stderr, "=> Stream == NULL!\n");
+        fflush(stderr);
+        return -1;
+      }
+      uint64_t r_streamid = SSL_get_stream_id(rstream);
+      uint64_t p_streamid = SSL_get_stream_id(pstream);
+      if (nghttp3_conn_bind_qpack_streams(h3conn, p_streamid, r_streamid)) {
+        printf("nghttp3_conn_bind_qpack_streams failed!\n");
+        return -1;
+      }
+      printf("control: NONE enc %ld dec %ld\n", p_streamid, r_streamid);
+      add_id(SSL_get_stream_id(rstream), rstream, h3ssl);
+      add_id(SSL_get_stream_id(pstream), pstream, h3ssl);
+      h3ssl->has_uni = 1;
+      return 0;
+    }
+  }
+  /* Well trying... */
+  if (numitem <= 1) {
+    return 0; /* Assume nothing for the moment */
+  }
+
+  /* Process the other stream */
+  for (int i = 1; i < numitem; i++) {
+    item++;
+    if (item->revents & SSL_POLL_EVENT_R) {
+      /* try to read */
+      int ret = 0;
```

Comment:
You're re-declaring the ret variable here from what you have on line 139.  Should probably reuse the variable or name this something different

Reply:
fixed.